vaadin-spring/src/main/java/com/vaadin/flow/spring/security/AuthenticationContext.java

```diff
@@ -140,6 +147,165 @@ public void logout() {
         });
     }
 
+    /**
+     * Gets the authorities granted to the current authenticated user.
+     *
+     * @return an unmodifiable collection of {@link GrantedAuthority}s or an
+     *         empty collection if there is no authenticated user.
+     */
+    public Collection<? extends GrantedAuthority> getGrantedAuthorities() {
+        return getAuthentication().filter(Authentication::isAuthenticated)
+                .map(Authentication::getAuthorities)
+                .orElse(Collections.emptyList());
+    }
+
+    /**
+     * Checks whether the current authenticated user has any of the given roles.
+     *
+     * @param roles
+     *            a collection containing at least one role.
+     * @return {@literal true} if the user holds at least one of the given
+     *         roles, otherwise {@literal false}.
+     * @throws IllegalArgumentException
+     *             if the given collection is empty.
+     */
+    public boolean hasAnyRole(Collection<String> roles) {
+        return hasAnyRole(roles.stream());
+    }
+
+    /**
+     * Checks whether the current authenticated user has any of the given roles.
+     *
+     * @param roles
+     *            an array containing at least one role.
+     * @return {@literal true} if the user holds at least one of the given
+     *         roles, otherwise {@literal false}.
+     * @throws IllegalArgumentException
+     *             if the given array is empty.
+     */
+    public boolean hasAnyRole(String... roles) {
+        return hasAnyRole(Stream.of(roles));
+    }
+
+    private boolean hasAnyRole(Stream<String> roles) {
+        return hasAnyAuthority(roles.map(role -> getRolePrefix() + role));
```

Comment:
```suggestion
        var rolePrefix = getRolePrefix();
        return hasAnyAuthority(roles.map(role -> rolePrefix + role));
```

hasAnyRole and hasAllRoles can be quite the hot-path; therefore I would reduce the amount of calculation needed to re-create the role prefix multiple times in the same request/check - especially with hundreds of roles..

While thinking about hot-path: Is the usage of stream here really necessary? (Up for others to decide)

Reply:
That was a good catch, I hadn't noticed it. I've now rewritten the checks to be more efficient.